backend/clustering_runner.py

```
@@ -0,0 +1,233 @@
+                "singletons": len(result["singletons"]),
+            }
+
+        update_cluster_job(
+            project_id,
+            job_id,
+            status="succeeded",
+            finished_at=datetime.now(timezone.utc),
+            stats=stats,
+        )
+    except Exception as exc:  # pragma: no cover - exercised in integration
+        logger.exception("Clustering job %s failed: %s", job_id, exc)
```

Ruff (0.14.8)

Redundant exception object included in `logging.exception` call

(TRY401)